includes/Modules/Analytics_4/Report/Request.php

```diff
@@ -380,4 +388,123 @@ protected function parse_dimension_filter( $dimension_name, $dimension_value ) {
 		return $filter_expression;
 	}
 
+	/**
+	 * Parses metric filters and returns a filter expression that should be added to the report request.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Data_Request $data Data request object.
+	 * @return Google_Service_AnalyticsData_FilterExpression The filter expression to use with the report request.
+	 */
+	protected function parse_metric_filters( Data_Request $data ) {
+		$expressions = array();
+
+		if ( is_array( $data['metricFilters'] ) ) {
+			foreach ( $data['metricFilters'] as $key => $value ) {
+				$expressions[] = $this->parse_metric_filter( $key, $value );
+			}
+		}
+
+		if ( ! empty( $expressions ) ) {
+			$filter_expression_list = new Google_Service_AnalyticsData_FilterExpressionList();
+			$filter_expression_list->setExpressions( array_filter( $expressions ) );
+
+			$metric_filters = new Google_Service_AnalyticsData_FilterExpression();
+			$metric_filters->setAndGroup( $filter_expression_list );
+
+			return $metric_filters;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Parses and returns a single metric filter.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $metric_name The metric name.
+	 * @param mixed  $metric_value The metric filter settings.
+	 * @return Google_Service_AnalyticsData_FilterExpression The filter expression instance.
+	 */
+	protected function parse_metric_filter( $metric_name, $metric_value ) {
+		// Use the numeric filter type by default.
+		$filter_type = 'numericFilter';
+		if ( isset( $metric_value['filterType'] ) ) {
+			// If there is the filterType property, use the explicit filter type then.
+			$filter_type = $metric_value['filterType'];
+		}
+
+		if ( 'numericFilter' === $filter_type ) {
+			if ( ! isset( $metric_value['operation'] ) && ! isset( $metric_value['value'] ) ) {
```

Comment:
Hmm, looks like this `&&` should be an `||`, as we need both `operation` and `value`:

```suggestion
			if ( ! isset( $metric_value['operation'] ) || ! isset( $metric_value['value'] ) ) {
```